HealthcareApp/src/Healthcare.Application/Appointments/Commands/Book/BookAppointmentCommandHandler.cs

```diff
@@ -0,0 +1,60 @@
+using Application.Abstractions;
+using Application.Abstractions.Decorators;
+using Domain.Entities;
+using Healthcare.Application.Appointments.Commands.Book;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using static Domain.Constants.AppointmentStatusConstants;
+
+namespace Healthcare.Application.Appointments.Commands;
+
+public class BookAppointmentCommandHandler : IRequestHandler<BookAppointmentCommand, bool>
+{
+    private readonly IUserManagerDecorator<ApplicationUser> _userManager;
+    private readonly IAppointmentRepository _appointmentRepository;
+
+
+    public BookAppointmentCommandHandler(IUserManagerDecorator<ApplicationUser> userManager,
+        IAppointmentRepository appointmentRepository)
+    {
+        _userManager = userManager;
+        _appointmentRepository = appointmentRepository;
+    }
+
+    public async Task<bool> Handle(BookAppointmentCommand request, CancellationToken cancellationToken)
+    {
+        var users = await _userManager.GetAllUsersAsync();
+
+        var doctor = await users.FirstOrDefaultAsync(x => x.Id == request.DoctorId, cancellationToken);
+        if (doctor == null) throw new Exception("Doctor not found");
+
+        var patient = await users.FirstOrDefaultAsync(x => x.Id == request.DoctorId, cancellationToken);
+        if (patient == null) throw new Exception("Patient not found");
+
+        var endTime = request.StartTime.Add(TimeSpan.FromMinutes(15));
+
+
+        var appointments = await _appointmentRepository.GetAllAppointmentsAsync();
+        var existingAppointment = appointments.FirstOrDefault<Appointment>(a =>
+            a.DoctorId == request.DoctorId.ToString() &&
+            a.AppointmentDate == request.AppointmentDate.Date &&
+            ((a.StartTime <= request.StartTime && a.EndTime > request.StartTime) ||
+             (a.StartTime < endTime && a.EndTime >= endTime)));
+        
+        if (existingAppointment != null) throw new Exception("Appointment slot already booked");
+
+        var appointment = new Appointment
+        {
+            DoctorId = request.DoctorId.ToString(),
+            PatientId = request.PatientId.ToString(),
+            AppointmentDate = request.AppointmentDate.Date,
+            StartTime = request.StartTime,
+            EndTime = endTime,
+            Status = Requested.ToString()
+        };
```

Comment:
I suggest you using any mapping library instead of manual mapping.
I recommend AutoMapper since it's the most popular mapping library and would probably be useful for your future job